src/pal/src/map/map.cpp

```diff
@@ -2241,7 +2241,9 @@ MAPmmapAndRecord(
     PAL_ERROR palError = NO_ERROR;
     LPVOID pvBaseAddress = NULL;
 
-    pvBaseAddress = mmap(addr, len, prot, flags, fd, offset);
+    off_t adjust = offset & (VIRTUAL_PAGE_MASK);
+
+    pvBaseAddress = mmap(static_cast<char *>(addr) - adjust, len + adjust, prot, flags, fd, offset - adjust);
```

Comment:
If I understand it correctly, when the addr is not aligned to the page size, we end up mapping some sections to virtual address space multiple times, possibly with different protection settings. So if there was a R/W section next to an executable section, we would create a potential security hole that would allow an attacker to modify the executable code through the adjacent R/W mapping.